pkgs/development/libraries/boost/generic.nix

```diff
@@ -11,7 +11,7 @@
 , enableShared ? !(hostPlatform.libc == "msvcrt") # problems for now
 , enableStatic ? !enableShared
 , enablePython ? hostPlatform == buildPlatform
-, enableNumpy ? false
+, enableNumpy ? enablePython && stdenv.lib.versionAtLeast version "1.65"
```

Comment:
Good points in https://github.com/NixOS/nixpkgs/pull/33359/files#r15934008 . How does this look?

Reply:
This more directly implements 19dbfb6's description "In order to manipulate Python arrays numpy is needed from boost 1.65 on."